File: src/app/shared/models/product.ts
export class Product{
    id?: number;
    name: string;
    categoryId: number;
    price: number;
    description: string;
    imageUrl: string;
    ratings: number;

    constructor(name:string, categoryId:number, price:number, description:string, imageUrl:string, ratings:number){
        this.name = name;
        this.categoryId = categoryId;
        this.price = price;
        this.description = description;
        this.imageUrl = imageUrl;
        this.ratings = ratings;
    }

}

File: api/Angular-Shop-Example-API/Angular-Shop-Example-API/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Angular_Shop_Example_API.Models
{

  [Table("tbl_products")]
  public class Product
  {
    public int id { get; set; }
    public string name { get; set; }
    public int price { get; set; }
    public string description { get; set; }
    public string imageUrl { get; set; }
    public int? categoryId { get; set; }

  }
}
